Panic while trying to join a manager via API

Using the latest engine-api, when I try to join a manager node and accept that node via API the initial manager panics.

```
time="2016-06-15T16:48:45.617828929Z" level=debug msg="Calling GET /nodes?filters=%7B%22id%22%3A%7B%2278b9maaib8bxs5qfqp6wksjxn%22%3Atrue%7D%7D" 
time="2016-06-15T16:48:45.619206357Z" level=debug msg="Calling POST /nodes/78b9maaib8bxs5qfqp6wksjxn/update?version=13" 
time="2016-06-15T16:48:45.619273918Z" level=debug msg="form data: {\"Availability\":\"active\",\"Membership\":\"accepted\",\"Role\":\"manager\"}" 
time="2016-06-15T16:48:45.628119487Z" level=debug msg="certificate issued" method="(*Server).signNodeCert" module=ca node.id=78b9maaib8bxs5qfqp6wksjxn node.role=MANAGER 
time="2016-06-15T16:48:45.980510569Z" level=debug method="(*Dispatcher).Tasks" node.id=78b9maaib8bxs5qfqp6wksjxn node.session=33wu9m0upgxt7rggbbwb7sxqn 
time="2016-06-15T16:48:45.988568293Z" level=error msg="failed to add member" error="grpc: target is unspecified" method="(*Node).Join" node.id=78b9maaib8bxs5qfqp6wksjxn 
time="2016-06-15T16:48:46.144787544Z" level=warning msg="sending message to an unrecognized member ID 5ceffd0cfdabb410" 
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x28 pc=0x1246d9c]

goroutine 1111 [running]:
panic(0x1aba540, 0xc82000e100)
	/usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/docker/swarmkit/manager/state/raft.(*Node).sendToMember(0xc820bd6ea0, 0xc821342e70, 0x8, 0x5ceffd0cfdabb410, 0x562c6098b786ca71, 0x2, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/docker/docker/vendor/src/github.com/docker/swarmkit/manager/state/raft/raft.go:850 +0x81c
created by github.com/docker/swarmkit/manager/state/raft.(*Node).send
	/go/src/github.com/docker/docker/vendor/src/github.com/docker/swarmkit/manager/state/raft/raft.go:806 +0x2b4
```

Here's an excerpt of the client code (perhaps I'm doing something wrong, but if so, the daemon probably shouldn't panic.)

```go
        options := types.NodeListOptions{} 
        nodes, err := primaryEngineClient.NodeList(context.TODO(), options)
        if err != nil {
                return err
        }
        // Just find the first available manager
        swarmManagerURL := ""
        for _, node := range nodes {
                if node.ManagerStatus != nil && node.ManagerStatus.Addr != "" {
                        swarmManagerURL = node.ManagerStatus.Addr
                        break
                }
        }
        if swarmManagerURL == "" {
                return fmt.Errorf("Something went wrong - couldn't find any existing manager nodes")
        }
        joinReq := swarm.JoinRequest{
                RemoteAddrs: []string{swarmManagerURL},
                Manager:     manager,
        }       
        err = joiningEngineClient.SwarmJoin(context.TODO(), joinReq)
        if err != nil {
                // ~Expected :-(
                // "Timeout reached before node was joined. Your cluster settings may be preventing this node from automatically joining. To accept this node into cluster run `docker node accept $ID` in an existing cluster manager"
                log.Debugf("Join result: %s", err)
                
                // Determine what this node identifies itself as
                info, err := joiningEngineClient.Info(context.TODO())
                if err != nil {
                        return err
                }
                newNodeID := info.Swarm.NodeID

                // Get the new node, and accept it
                args := filters.NewArgs()
                args.Add("id", newNodeID)
                options = types.NodeListOptions{
                        Filter: args,
                }       
                nodes, err := primaryEngineClient.NodeList(context.TODO(), options)
                if err != nil {
                        return err
                }       
                if len(nodes) >= 1 {
                        nodes[0].Spec.Membership = swarm.NodeMembershipAccepted
                        log.Debug("Accepting join request")
                        err := primaryEngineClient.NodeUpdate(context.TODO(), newNodeID, nodes[0].Version, nodes[0].Spec)
                        if err != nil {
                                return err
                        }       
                        log.Debug("Success")
                        return nil
                }       
                return fmt.Errorf("Unable to locate pending node")
        } else {
                log.Info("Join completed without requiring accept")
        }       
        
        return nil
```

This code reports "Success" but then the daemon panics.

--- a/manager/manager.go
+++ b/manager/manager.go
@@ -3,6 +3,7 @@ package manager
 import (
 	"crypto/x509"
 	"encoding/pem"
+	"errors"
 	"fmt"
 	"net"
 	"os"
@@ -100,6 +101,10 @@ func New(config *Config) (*Manager, error) {
 	}
 
 	tcpAddr := config.ProtoAddr["tcp"]
+
+	if tcpAddr == "" {
+		return nil, errors.New("no tcp listen address or listener provided")
+	}
 
 	listenHost, listenPort, err := net.SplitHostPort(tcpAddr)
 	if err == nil {
@@ -664,7 +669,7 @@ func (m *Manager) rotateRootCAKEK(ctx context.Context, clusterID string) error {
 	return s.Update(func(tx store.Tx) error {
 		cluster = store.GetCluster(tx, clusterID)
 		if cluster == nil {
-			return fmt.Errorf("cluster not found")
+			return fmt.Errorf("cluster not found: %s", clusterID)
 		}
 		cluster.RootCA.CAKey = finalKey
 		return store.UpdateCluster(tx, cluster)

